horovod/common/response_cache.cc

```diff
@@ -0,0 +1,223 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =============================================================================
+
+#include "response_cache.h"
+
+namespace horovod {
+namespace common {
+
+void ResponseCache::clear() {
+  capacity_ = 0;
+  bits_outdated_ = false;
+  cache_.clear();
+  cache_iters_.clear();
+  tensor_name_to_bit_.clear();
+}
+
+void ResponseCache::set_capacity(uint32_t capacity) {
+  // Clear cache in case set_capacity is called multiple times if autotuning.
+  // Only clear if capacity is modified.
+  if (capacity != capacity_) {
+    this->clear();
+  }
+
+  capacity_ = capacity;
+  cache_iters_.reserve(capacity);
+}
+
+uint32_t ResponseCache::capacity() const { return capacity_; }
+
+size_t ResponseCache::current_size() const { return cache_.size(); }
```

Comment:
CLion is telling me this method is not used, remove?

Reply:
Removed.